frontend/src/Components/RatingSwitch.tsx

```diff
@@ -52,27 +51,17 @@ export default function RatingSwitch(props: RatingSwitchProps) {
                 });
         }
 
+        const [popupEl, ratingEl] = [popupRef.current, ratingRef.current];
+
         const rect = ratingEl.getBoundingClientRect();
-        const x = rect.x;
         const y = rect.y + document.documentElement.scrollTop || 0;
-        const [w, h] = [ratingEl.clientWidth, ratingEl.clientHeight];
-        const [pw, ph] = [popupEl.clientWidth, popupEl.clientHeight];
-
-        let ny = y + h + 8;
-        if (ny + ph > document.documentElement.scrollHeight) {
-            ny = y - 8 - ph;
-        }
-        let nx = x;
-        if (nx + 8 + pw > document.documentElement.scrollWidth) {
-            if (x+w+8>pw) {
-                nx = x + w - pw;
-            } else {
-                nx = 8;
-            }
-        }
+        const rh = ratingEl.clientHeight;
+        const ph = popupEl.clientHeight;
 
-        popupEl.style.left = (nx) + 'px';
-        popupEl.style.top = (ny) + 'px';
+        const isTotalHeightMoreThanPageHeight  = y + rh + ph > document.documentElement.scrollHeight;
+        isTotalHeightMoreThanPageHeight
+            ? popupEl.style.bottom = 30 + 'px'
+            : popupEl.style.top = 30 + 'px';
```

Comment:
nit
```suggestion
        if(isTotalHeightMoreThanPageHeight){
            popupEl.style.bottom = '30px'
        } else {
            popupEl.style.top = 30 + '30px';
        }
```

Reply:
@4vanger should we add `no-ternary` to eslint?

Reply:
Oh, I see. `30 + '30px'`. 

 It is allowed to use ternary?

Reply:
After these changes, we have a huge gap. Is it ok?

<img width="370" alt="Screenshot 2023-04-13 at 19 23 27" src="https://user-images.githubusercontent.com/3425688/231824582-d744e1d5-5c30-453e-9b48-498ad6942fb2.png">

Reply:
> Oh, I see. `30 + '30px'`.
nope, it was my mistype. fixed.

> It is allowed to use ternary?

Ternary is perfectly fine - just assignment in ternary looks weird as for me. We don't have strict rules about codestyles unfortunately.